Sources/SwiftParser/TokenConsumer.swift

```diff
@@ -79,7 +79,7 @@ extension TokenConsumer {
   ///
   /// - Parameter kind: The kind of token to consume.
   /// - Returns: A token of the given kind.
-  public mutating func eat(_ kind: RawTokenKind) -> Token {
+  public mutating func eatWithoutRecovery(_ kind: RawTokenKind) -> Token {
```

Comment:
Let's move this utility into Parser and Lookup respectively if can no longer be shared. The new name doesn't read any better to me.